src/compiler/checker.ts

```diff
@@ -31353,6 +31356,42 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
+    function checkGrammarRegularExpressionLiteral(node: RegularExpressionLiteral) {
+        const sourceFile = getSourceFileOfNode(node);
+        if (!hasParseDiagnostics(sourceFile)) {
+            let lastError: DiagnosticWithLocation | undefined;
+            const scanner = createScanner(

```

Comment:
It might be good to cache a single scanner here and reset the state instead.

Reply:
Perhaps, though we create a lot of single-use scanners during check already due to `getSpanOfTokenAtPosition` and `scanTokenAtPosition`.

Reply:
Yeah, I've been trying to figure out a good way to do this more generally, but this isn't new.